File: IntelligentRecruitment-UI/src/app/models/Dto/Application/ApplicationResponse.ts
import {ApplicationStatus} from '../../Enum/ApplicationStatus';

export interface ApplicationResponse {
  id:number;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  llmScore: number; // BigDecimal côté Java -> number côté TS
  llmAnalysis: string;
  cvFilePath: string;
  createdAt: string; // LocalDateTime -> string (ISO date)
  status: ApplicationStatus;
}

File: IntelligentRecruitment_microservice/application-service/src/main/java/com/fedicode/applicationservice/Dto/ApplicationResponse.java
package com.fedicode.applicationservice.Dto;

import com.fedicode.applicationservice.Entity.Application;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
public class ApplicationResponse {
    int id;
    String firstName;
    String lastName;
    String email;
    String phone;
    BigDecimal llmScore;
    String llmAnalysis;
    String cvFilePath;
    LocalDateTime createdAt;
    Application.ApplicationStatus status;
}
